src/commands/remove/index.ts

```diff
@@ -25,35 +25,66 @@ registerCommand<PreHandlers>({
 			this.EarlyEnd.InvalidInput('No emote name provided');
 		}
 
-		const emote = (await gql.CurrentEnabledEmotes(EmoteSet())).find(
-			(emote) => emote.name === ctx.input[0],
-		);
+		ctx.input = ctx.input.reduce((args: string[], arg: string) => {
+			if (!args.includes(arg))
+				args.push(arg)
+			return args
+		}, [])
 
-		if (!emote) {
-			return {
-				Success: false,
-				Result: 'Could not find that emote',
-			};
+		const emotes: (never | EnabledEmote)[] = []
+		for (const emote of await gql.CurrentEnabledEmotes(EmoteSet())) {
+			const emoteIdx: number = ctx.input.indexOf(emote.name)
+
+			if (emoteIdx === -1)
+				continue
+
+			emotes.push(emote)
+			ctx.input.splice(emoteIdx, 1)
+
+			if (!ctx.input.length)
+				break
 		}
 
-		try {
-			await gql.ModifyEmoteSet(EmoteSet(), ListItemAction.REMOVE, emote.id);
-		} catch (error) {
-			ctx.Log('info', '7TV - Failed to remove emote', error);
-			return {
-				Success: false,
-				Result: `Error removing the emote => ${emote.name}`,
-			};
+		if (ctx.input.length)
+			ctx.channel.say(`Could not find the following emote${ctx.input.length > 1 ? 's' : ''}: ${ctx.input.join(' ')}`)
+
+		const failed: (undefined | string)[] = (await Promise.all(
+			emotes.map(async emote => {
+				try {
+					await gql.ModifyEmoteSet(EmoteSet(), ListItemAction.REMOVE, emote.id)
+				}
+				catch (error) {
+					ctx.Log('info', '7TV - Failed to remove emote', error)
+					return emote.name
+				}
+			})
+		)).filter(Boolean)
+
+		type out = {
+			Success: boolean
+			Result: string
 		}
 
-		return {
+		return (function (this: out) {
```

Comment:
Could you figure out  something that doesn't require this weird function thing or that call method please.

You can remove the out type, and instead use [this](https://github.com/JoachimFlottorp/Melonbot/blob/main/src/Models/Command.ts#L50)

Reply:
I mean ts doesn't complain, it's better than the alternatives, it's more or less how object literals actually work under the hood, while allowing you the freedom of conditions, a `new function()` directly would be better/preferred, but ts was complaining.
I don't believe there is a better option tbh.